mars/services/task/execution/ray/fetcher.py

```diff
@@ -23,17 +24,28 @@
 _FetchInfo = namedtuple("FetchInfo", ["key", "object_ref", "conditions"])
 
 
+def _query_object_with_condition(o, conditions):
+    try:
+        return o.iloc[conditions]
+    except AttributeError:
+        return o[conditions]
+
+
 @register_fetcher_cls
 class RayFetcher(Fetcher):
     name = "ray"
     required_meta_keys = ("object_refs",)
 
     def __init__(self, **kwargs):
-        _make_query_function_remote()
-
         self._fetch_info_list = []
         self._no_conditions = True
 
+    @staticmethod
+    @functools.lru_cache(maxsize=None)
```

Comment:
`maxsize=None` is unnecessary

Reply:
The default `max_size` of lru_cache is 128. if the `max_size > 0`, then the cache implementation is more complicated. Specify the `maxsize=None` to use a simpler cache implementation.

Reply:
`_remote_query_object_with_condition` has no params, so I guess there will be only one cache item. Skip specifying will be OK?

Reply:
> `_remote_query_object_with_condition` has no params, so I guess there will be only one cache item. Skip specifying will be OK?

Since the `_remote_query_object_with_condition` has no params, the cache item should be always 1. Specify a None maxsize to make it faster because we don't need a maxsize.

Reply:
Comments added. Thanks.